amlb/datasets/file.py

```diff
@@ -167,6 +245,59 @@ def _get_metadata(self, prop):
         return meta[prop]
 
 
+class DatasetWithAuxiliaryData:
```

Comment:
must extend `Dataset` if you want to expose this.

Actually, I think the right interface would be sth like:
```python
# in data.py

class Datasplit(ABC):
    ...

    @property
    def has_auxiliary_data(self) -> bool:
        pass
       
    @property
    def auxiliary_data(self) -> List[AuxData]:
        pass
        
class AuxData(ABC):
    # does it need a name property? a type/category/labels?

    @property
    def path(self) -> str:
        pass

    @property
    @abstractmethod
    def data(self) -> DF:
        """
        :return: the auxiliary data as a pandas DataFrame.
        """
        pass

    @profile(logger=log)
    def release(self, properties=None):
        clear_cache(self, properties)
```
then you can implement the logic for files to fulfill the contract of this interface.

Reply:
> Having auxiliary data in Datasplit seems unnecessary

why? you were planning to have `dataset.train_auxiliary_path` and `dataset.test_auxiliary_path`.
I'm suggesting that for you it would then be accessed as `dataset.train.auxiliary_data[0].path` and `dataset.test.auxiliary_data[0].path`: it's much more consistent with the current interfaces.

Everything that is exposed to frameworks must be defined in one of the interfaces defined in `data.py`.

> This is identical to DatasetWithAuxiliaryData

no, this is not identical: you're creating a class that doesn't extend anything and you add features that are not visible in the contract defined by the interfaces: those are the only references for all data providers.

to sum up, a dataset has currently 2 splits: `train` and `test`.
you can have aux data for each of those, therefore auxdata should be a property of `Datasplit`: if you add the auxdata directly to the Dataset instance instead of Datasplit, you create an inconsistency.

Reply:
Makes sense. So do you think adding method to Dataset to add aux data to existing data split is a good idea? Or we would need to have loading of dataset and aux data at the same time then construct the Dataset object. This would require refactor of DataLoader.load